diff --git a/src/core/transport/http/sender/libcurl/axis2_libcurl.c b/src/core/transport/http/sender/libcurl/axis2_libcurl.c
index 87635a9..d851955 100644
--- a/src/core/transport/http/sender/libcurl/axis2_libcurl.c
+++ b/src/core/transport/http/sender/libcurl/axis2_libcurl.c
@@ -111,7 +111,7 @@ axis2_libcurl_send(
     axis2_char_t *buffer = NULL;
     unsigned int buffer_size = 0;
     int content_length = -1;
-    axis2_char_t *content_type;
+    axis2_char_t *content_type = NULL;
     axis2_char_t *content_len = AXIS2_HTTP_HEADER_CONTENT_LENGTH_;
     const axis2_char_t *char_set_enc = NULL;
     axis2_char_t *content = AXIS2_HTTP_HEADER_CONTENT_TYPE_;
@@ -140,7 +140,7 @@ axis2_libcurl_send(
 
     handler = data->handler;
     curl_easy_reset(handler);
-    curl_easy_setopt(handler, CURLOPT_ERRORBUFFER, &data->errorbuffer);
+    curl_easy_setopt(handler, CURLOPT_ERRORBUFFER, data->errorbuffer);
     headers = curl_slist_append(headers, AXIS2_HTTP_HEADER_USER_AGENT_AXIS2C);
     headers = curl_slist_append(headers, AXIS2_HTTP_HEADER_ACCEPT_);
     headers = curl_slist_append(headers, AXIS2_HTTP_HEADER_EXPECT_);
