diff --git a/samples/client/Makefile.am b/samples/client/Makefile.am
index 763c375..23b0038 100644
--- a/samples/client/Makefile.am
+++ b/samples/client/Makefile.am
@@ -21,7 +21,7 @@ AUTOMAKE_OPTIONS = foreign
 bin_PROGRAMS = KolibreSampleClient
 KolibreSampleClient_SOURCES = Main.cpp Input.cpp
 
-KolibreSampleClient_LDFLAGS = @LOG4CXX_LIBS@ @GLIB_LIBS@ -L$(top_builddir)/src/.libs -lkolibre-clientcore -lboost_regex -lboost_filesystem
+KolibreSampleClient_LDFLAGS = @LOG4CXX_LIBS@ @GLIB_LIBS@ -L$(top_builddir)/src/.libs -lkolibre-clientcore
 KolibreSampleClient_CPPFLAGS = @LOG4CXX_CFLAGS@ @GLIB_CFLAGS@ -I$(top_srcdir)/src
 
 copy-files:
diff --git a/src/Makefile.am b/src/Makefile.am
index b76307a..4c7f796 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -50,7 +50,7 @@ lib_LTLIBRARIES = libkolibre-clientcore.la
 
 libkolibre_clientcore_la_SOURCES = $(SRCS)
 libkolibre_clientcore_la_LIBADD = $(top_builddir)/src/Settings/libsettings.la
-libkolibre_clientcore_la_LDFLAGS = -version-info $(VERSION_INFO) @LOG4CXX_LIBS@ @LIBKOLIBRENAVIENGINE_LIBS@ @LIBKOLIBREPLAYER_LIBS@ @LIBKOLIBRENARRATOR_LIBS@ @LIBKOLIBREDAISYONLINE_LIBS@ @LIBKOLIBREXMLREADER_LIBS@ @LIBKOLIBREAMIS_LIBS@ @DBUS_LIBS@ @DBUSGLIB_LIBS@
+libkolibre_clientcore_la_LDFLAGS = -version-info $(VERSION_INFO) @LOG4CXX_LIBS@ @LIBKOLIBRENAVIENGINE_LIBS@ @LIBKOLIBREPLAYER_LIBS@ @LIBKOLIBRENARRATOR_LIBS@ @LIBKOLIBREDAISYONLINE_LIBS@ @LIBKOLIBREXMLREADER_LIBS@ @LIBKOLIBREAMIS_LIBS@ @DBUS_LIBS@ @DBUSGLIB_LIBS@ -lboost_regex -lboost_filesystem -lboost_system
 libkolibre_clientcore_la_CPPFLAGS = @LOG4CXX_CFLAGS@ @LIBKOLIBRENAVIENGINE_CFLAGS@ @LIBKOLIBREPLAYER_CFLAGS@ @LIBKOLIBRENARRATOR_CFLAGS@ @LIBKOLIBREDAISYONLINE_CFLAGS@ @LIBKOLIBREXMLREADER_CFLAGS@ @LIBKOLIBREAMIS_CFLAGS@ @DBUS_CFLAGS@ @DBUSGLIB_CFLAGS@
 
 EXTRA_DIST = CommandQueue2/CommandQueue.h \
