Index: git/samples/client/Makefile.am
===================================================================
--- git.orig/samples/client/Makefile.am
+++ git/samples/client/Makefile.am
@@ -24,10 +24,5 @@ KolibreSampleClient_SOURCES = Main.cpp I
 KolibreSampleClient_LDFLAGS = @BOOST_LDFLAGS@ @LOG4CXX_LIBS@ -L$(top_builddir)/src/.libs -lkolibre-clientcore -lboost_regex
 KolibreSampleClient_CPPFLAGS = @BOOST_CPPFLAGS@ @LOG4CXX_CFLAGS@ -I$(top_srcdir)/src
 
-all-local:
-	cp $(top_srcdir)/samples/client/run $(top_builddir)/samples/client
-	cp $(top_srcdir)/samples/client/run.sh $(top_builddir)/samples/client
-	cp $(top_srcdir)/samples/client/log4cxx.conf $(top_builddir)/samples/client
-
 EXTRA_DIST = run.sh \
 			 Input.h
